swarm/pss/client/client.go

```diff
@@ -47,21 +47,22 @@ type Client struct {
 // implements p2p.MsgReadWriter
 type pssRPCRW struct {
 	*Client
-	topic    pss.Topic
+	topichex string
```

Comment:
Is it necessary to rename things to "...hex"? Wouldn't it be better to just implement the JSON marshaler functions to ensure these things are encoded as hex when necessary?

Reply:
I'd like to keep it like this for clarity. It relates to how it's written in the test. Please indulge me.

Reply:
I don't see how it provides clarity though, how is it better than `topic pss.Topic` (which, in my opinion, provides more clarity on what the field is).

Reply:
Because `topic` is an actual type, and is used as variable name for that type everywhere else in the code. Whereas `topichex` suggests it's a hex representation of the same, and used everywhere else in the code in such representation.